src/libcore/iter/iterator.rs

```diff
@@ -258,6 +258,43 @@ pub trait Iterator {
         None
     }
 
+    /// Returns an iterator adaptor that goes over the original iterator,
+    /// skipping `step` elements at a time.
```

Comment:
I'm struggling with these docs. It says that it *skips* `step` elements at a time, but in the example, `step == 2` but it's only skipping one element at a time.